### PR TITLE
feat(ecr): Ensure ECR repositories have tag immutability configured

### PR DESCRIPTION
**Pull request by MarioRgzLpz**
### Context

Added new logic to check whether a ECR repository has tag immutability enabled. The check fails if a private ECR repository has tag immutability disabled and passes if tag immutability is set to IMMUTABLE.
Amazon ECR Tag Immutability enables customers to rely on the descriptive tags of an image as a reliable mechanism to track and uniquely identify images. An immutable tag is static, which means each tag refers to a unique image. This improves reliability and scalability as the use of a static tag will always result in the same image being deployed. When configured, tag immutability prevents the tags from being overridden, which reduces the attack surface.

### Description

Change `Repository` model in `ecr_service` to add a new atributte `immutability` for the check. Added new check `ecr_repositories_immutability_tag_configured` with respective unit tests.

### Checklist

- Are there new checks included in this PR? Yes
    - If so, do we need to update permissions for the provider? No
- [x] Review if the code is being covered by tests.
- [x] Review if code is being documented following this specification https://github.com/google/styleguide/blob/gh-pages/pyguide.md#38-comments-and-docstrings
- [x] Review if backport is needed.

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prowler/providers/aws/services/ecr/ecr_repositories_immutability_tag_configured/ecr_repositories_immutability_tag_configured.metadata.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
+{
+  "Provider": "aws",
+  "CheckID": "ecr_repositories_immutability_tag_configured",
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/ecr/ecr_repositories_immutability_tag_configured/ecr_repositories_immutability_tag_configured.metadata.json]:
```suggestion
  "CheckID": "ecr_private_repository_tag_immutability",
```

File: prowler/providers/aws/services/ecr/ecr_repositories_immutability_tag_configured/ecr_repositories_immutability_tag_configured.py
Comment (MrCloudSec) [inline prowler/providers/aws/services/ecr/ecr_repositories_immutability_tag_configured/ecr_repositories_immutability_tag_configured.py:1]: You should only check private repositories.

File: prowler/providers/aws/services/ecr/ecr_service.py [hunk 2]
```diff
@@ -47,6 +47,8 @@ def _describe_registries_and_repositories(self, regional_client):
                                 scan_on_push=repository["imageScanningConfiguration"][
                                     "scanOnPush"
                                 ],
+                                immutability=repository["imageTagMutability"]
+                                == "IMMUTABLE",
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/ecr/ecr_service.py]:
```suggestion
                                immutability=repository.get("imageTagMutability", "MUTABLE")
```

Please, compare the strings in the check.

File: prowler/providers/aws/services/ecr/ecr_repositories_immutability_tag_configured/ecr_repositories_immutability_tag_configured.py
Comment (MarioRgzLpz) [reply prowler/providers/aws/services/ecr/ecr_repositories_immutability_tag_configured/ecr_repositories_immutability_tag_configured.py:1]: As far as I understand using [ECR](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/ecr.html) we are only returning the private repositories. To return the public repositories we should use [ECRPublic](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/ecr-public.html). Maybe Im missing something and we are indeed returning all repositories, in that case we should change some checks that right now are checking every repository including public ones.

**Comment by MarioRgzLpz** [comment]
@sergargar Requested changes added in commit -> [6e3083d](https://github.com/prowler-cloud/prowler/pull/5144/commits/6e3083d4c16d05088ee54c850235784f47ed5edc)

File: prowler/providers/aws/services/ecr/ecr_repositories_immutability_tag_configured/ecr_repositories_immutability_tag_configured.py
Comment (MrCloudSec) [reply prowler/providers/aws/services/ecr/ecr_repositories_immutability_tag_configured/ecr_repositories_immutability_tag_configured.py:1]: You're right! Thanks for the info

**Review by MrCloudSec — APPROVED** [review]
Good job!

I changed the name again to follow the same pattern as the rest of ECR checks.

**Pull request merged**